### ISSUE TITLE
Support endpointslices - Prefer Same zone traffic to save costs

### ISSUE THREAD

**Issue by szuecs** [zalando/skipper#1446]
Right now in kubernetes dataclient we use endpoints to get lb pool members. These are flat ip port tuples.

In Kubernetes there is a new feature endpointslices that have topology which enables us to have endpoints for specific zones for example. With this we can optimize traffic costs, because same zone traffic costs less than cross zone traffic.
See also [KEP topology per endpoint](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/20190603-endpointslices#topology-per-endpoint) and [KEP API](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/20190603-endpointslices#endpointslice-api)

In a discussion in slack a user wrote:

> for us, the cost savings are huge. We've already implemented it in a few places in the application and it's saved about 5% of our total AWS bill.

5% is huge, and I think we to have have more priority for this

see also: 
- https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/
- https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/

Steps
- [x] integrate endpointsclies https://github.com/zalando/skipper/issues/2551
- [x] zone aware routing https://github.com/zalando/skipper/issues/2552

**Comment by szuecs** [comment]
Interesting KEP updates
https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2030-endpointslice-subsetting#proposal
https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2004-topology-aware-subsetting

**Comment by szuecs** [comment]
See also https://twitter.com/a_sykim/status/1413492899258400770?s=21 for changes in kubernetes 1.22

**Comment by szuecs** [comment]
https://github.com/kubernetes/enhancements/pull/4003 is an interesting change proposal within kubernetes

**Comment by szuecs** [comment]
A quick hack to enable endpointslices usage would be to have an annotation on ingress or routegroup to force service use instead of endpoints https://kubernetes.io/docs/concepts/services-networking/topology-aware-routing/

**Comment by AlexanderYastrebov** [comment]
> A quick hack to enable endpointslices usage would be to have an annotation on ingress or routegroup to force service use instead of endpoints

I think we have a flag for that introduced by #2088

**Comment by szuecs** [comment]
@AlexanderYastrebov yes but by annotation would reduce the blast radius to test. If you have a cluster with a few pods and hug number of pods side-by-side than the few pods can easily be impaired.

Example you have a backend with 8 pods, 3 in zone A and B and 2 in C.
Now skipper-ingress has 4 pods, 1 in zone A and B and 2 in C.
NLB and other internal clients will spread the load across all 4 skipper pods and now skipper has 50% traffic in C and only 25% in A and B. Now let's assume we use zone aware routing in skipper, then you will get 50% traffic to C but you have 8 pods, so you serve 50% traffic with 25% of your pods. Would you think it's a great solution?
I don't think it is, but this would be the way how it works.

**Comment by szuecs** [comment]
We have the full chain implemented
